dspace-sword/src/main/java/org/purl/sword/client/ClientOptions.java

```diff
@@ -148,128 +155,122 @@ public class ClientOptions {
-                }
+        Option option;
 
-                if ("-verbose".equals(args[i])) {
-                    verbose = true;
-                }
+        option = Option.builder().longOpt("host").hasArg().build();
+        options.addOption(option);
 
-                if ("-cmd".equals(args[i])) {
-                    guiMode = false;
-                }
+        option = Option.builder().longOpt("port").hasArg().build();
+        options.addOption(option);
 
-                if ("-gui".equals(args[i])) {
-                    guiMode = true;
-                }
+        option = Option.builder("u").hasArg().build();
+        options.addOption(option);
 
-                if ("-host".equals(args[i])) {
-                    i++;
-                    proxyHost = args[i];
-                }
+        option = Option.builder("p").hasArg().build();
+        options.addOption(option);
 
-                if ("-port".equals(args[i])) {
-                    i++;
-                    proxyPort = Integer.parseInt(args[i]);
-                }
+        option = Option.builder().longOpt("href").hasArg().build();
+        options.addOption(option);
 
-                if ("-u".equals(args[i])) {
-                    i++;
-                    username = args[i];
-                }
+        option = Option.builder("t").hasArg().build();
+        options.addOption(option);
 
-                if ("-p".equals(args[i])) {
-                    i++;
-                    password = args[i];
-                }
+        option = Option.builder().longOpt("file").hasArg().build();
+        options.addOption(option);
 
-                if ("-href".equals(args[i])) {
-                    i++;
-                    href = args[i];
-                }
+        option = Option.builder().longOpt("filetype").hasArg().build();
+        options.addOption(option);
 
-                if ("-help".equals(args[i])) {
-                    // force the calling code to display the usage information
-                    return false;
-                }
+        option = Option.builder().longOpt("slug").hasArg().build();
+        options.addOption(option);
 
-                if ("-t".equals(args[i])) {
-                    i++;
-                    accessType = args[i];
-                }
-
-                if ("-file".equals(args[i])) {
-                    i++;
-                    filename = args[i];
-                }
+        option = Option.builder().longOpt("onBehalfOf").hasArg().build();
+        options.addOption(option);
 
-                if ("-filetype".equals(args[i])) {
-                    i++;
-                    filetype = args[i];
-                }
+        option = Option.builder().longOpt("formatNamespace").hasArg().build();
+        options.addOption(option);
 
-                if ("-slug".equals(args[i])) {
-                    i++;
-                    slug = args[i];
-                }
+        option = Option.builder().longOpt("checksumError").hasArg().build();
```

Comment:
In the original code, this doesn't appear to have an argument, so I think the `hasArg()` should be dropped here.

Reply:
Correct, thanks!